code/to_bsdd.py

```diff
@@ -197,39 +230,56 @@ def get_parent_of_pt(enum_type):
                 if item.stereotype == "QSET":
                     type_name = "real"
                     type_values = None
+                    kind_name = 'Single'
                 else:
                     ty, ty_arg = ty_ty_arg
                     if ty == "PropertyEnumeratedValue":
                         type_name = list(ty_arg.values())[0]
                         enum_types_by_name = [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:Class"] if c.name == type_name]
                         enum_types_by_name += [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:Enumeration"] if c.name == type_name]
                         type_values = [x.name for x in enum_types_by_name[0]/"ownedLiteral"]
-                    elif ty == "PropertySingleValue":
+                        kind_name = 'Single'
+                    else:
                         type_name = list(ty_arg.values())[0]
                         pe_types = [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:Class"] if c.name == type_name]
                         pe_types += [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:DataType"] if c.name == type_name]                    
                         pe_type = pe_types[0]
                         root_generalization = generalization(pe_type)
                         type_name = root_generalization.name.lower()
                         type_values = None
-                    else:
-                        print("Warning: %s.%s of type %s <%s> not mapped" % (item.name, nm, ty, ",".join(map(lambda kv: "=".join(kv), ty_arg.items()))))
-                        continue
+                        if ty == "PropertySingleValue":
+                            kind_name = 'Single'
+                        elif ty == "PropertyBoundedValue":
+                            di["Psets"][item.name]["Properties"][a.name]["Description"] = "This property in IFC stores a bounded value, meaning it has a maximum of two (numeric or descriptive) values assigned, the first value specifying the upper bound and the second value specifying the lower bound. Read the IFC documentation for more information."
+                            kind_name = 'Range'
+                        elif ty == "PropertyReferenceValue":
+                            di["Psets"][item.name]["Properties"][a.name]["Description"] = "This property in IFC takes as its value a reference to one of: IfcAddress, IfcAppliedValue, IfcExternalReference, IfcMaterialDefinition, IfcOrganization, IfcPerson, IfcPersonAndOrganization, IfcTable, IfcTimeSeries. Read the IFC documentation for more information."
+                            kind_name = 'Complex'
```

Comment:
Is this intentional?

```suggestion
                            kind_name = 'Reference'
```

I would keep this out of bSDD though. What are the implications otherwise of bSDD value constraints? It's a whole can of worms for very little value. These properties are basically not used.

Reply:
I wanted to have all properties from IFC in bSDD, so that we can prove they are not used. Complex values are not supported by software, so their restrictions don't matter anyway. 

The kind name allowed by bSDD is 'Single', 'Range, 'List', 'Complex' and 'ComplexList'.